src/Orleans/Core/ClusterClient.cs

```diff
@@ -133,12 +132,17 @@ public void Abort()
                 try
                 {
                     this.state = LifecycleState.Disposing;
-                    if (gracefully)
+
+                    if (this.runtimeClient is OutsideRuntimeClient outsideClient)
                     {
-                        Utils.SafeExecute(() => this.runtimeClient.Disconnect());
+                        if (gracefully)
+                        {
+                            Utils.SafeExecute(() => outsideClient.Disconnect());
+                        }
+
+                        Utils.SafeExecute(() => outsideClient.Reset(gracefully));
```

Comment:
why was the `Reset` method removed from the interface, forcing a need to downcast?

Reply:
The interface is for all `IRuntimeClient`, but `Reset(bool)` is only implemented in `OutsideRuntimeClient` - it would throw on `InsideRuntimeClient`

Reply:
It might make sense to have some kind of specific `ClusterClient` or an interface for the external runtime client.